app/services/arcgis_api/geocoder.rb

```diff
@@ -58,15 +57,23 @@ def find_address_candidates(magic_key)
 
     # Makes a request to retrieve a new token
     # it expires after 1 hour
-    # @return [String] the token
+    # @return [String] Auth token
     def retrieve_token!
-      body = request_token
-      @@token_expires_at = Time.zone.at(body['expires'])
-      @@token = body['token']
+      token_response_body = request_token
+      new_token = token_response_body['token']
+
+      Rails.cache.write(
+        API_TOKEN_CACHE_KEY, new_token,
+        expires_at: Time.zone.at(token_response_body['expires'])
+      )
+      new_token
```

Comment:
Yeah, they appear to be similar. I guess the difference is in how they handle missing keys, but not something that should be relevant here (hopefully!). I think the default behavior of `fetch_values` to raise on missing key may be a desirable-enough behavior, since it'd be very unexpected.